crates/pessimistic-proof/src/keccak.rs

```diff
@@ -38,6 +38,7 @@ pub fn keccak256(data: &[u8]) -> Digest {
 }
 
 /// Hashes the input items using a Keccak hasher with a 256-bit security level.
+/// Safety: This function should only be called with fixed-size items to avoid collisions.
```

Comment:
Maybe we could try to impose the constraint by types. Now we have `T: AsRef<[u8]>`. Maybe changing it to something like `T: AsRef<[u8; N]>` or `T: Borrow<[u8; N]>` or `T: Into<[u8; N]>` (not sure which one to pick here, maybe just try which one, if any, already compiles) would work. There would have to be a new generic param `const N: usize`.

Reply:
I think this wouldn't allow hashing items of different size, which we do for example here https://github.com/agglayer/agglayer/blob/e462db6b30a76807f2763a320f7a552c5181b1dd/crates/pessimistic-proof/src/bridge_exit.rs#L103

Reply:
> I think this wouldn't allow hashing items of different size, which we do for example here
> 
> https://github.com/agglayer/agglayer/blob/e462db6b30a76807f2763a320f7a552c5181b1dd/crates/pessimistic-proof/src/bridge_exit.rs#L103

For hashing fixed number of heterogeneous data item, we could just put all the items into a tuple and hash that. I think it gives the same result, does it not?

Reply:
Yes, as mentioned #234 there are secure ways to serialize data for hashing, but they are only required when hashing variable length data, which we don't do in our crates. So I think it's fine to keep using `keccak256_combine` with this safety note so that everyone is aware of the potential vulnerability.